worker/logger/logger_test.go

```diff
@@ -110,3 +109,35 @@ func (s *LoggerSuite) TestConfigOverride(c *gc.C) {
 	expected := "<root>=WARNING;test=TRACE"
 	s.waitLoggingInfo(c, expected)
 }
+
+type mockNotifyWatcher struct {
+	changes chan struct{}
+}
+
+func (m *mockNotifyWatcher) Kill() {}
+
+func (m *mockNotifyWatcher) Wait() error {
+	return nil
+}
+
+func (m *mockNotifyWatcher) Changes() watcher.NotifyChannel {
+	return m.changes
+}
+
+func (m *mockNotifyWatcher) change() {
+	m.changes <- struct{}{}
+}
+
+var _ watcher.NotifyWatcher = (*mockNotifyWatcher)(nil)
+
+type mockAPI struct {
+	watcher *mockNotifyWatcher
+	config  string
+}
+
+func (m *mockAPI) LoggingConfig(agentTag names.Tag) (string, error) {
+	return m.config, nil
+}
+func (m *mockAPI) WatchLoggingConfig(agentTag names.Tag) (watcher.NotifyWatcher, error) {
+	return m.watcher, nil
```

Comment:
we should set up the mock api with the expected agent and then assert here that the agent matches to make sure we are passing in the expected value, or just assert that it equals 42 as that's what was set up earlier